terraform/ec2/win/main.tf

```diff
@@ -34,66 +58,60 @@ resource "aws_instance" "cwagent" {
   ami                         = data.aws_ami.latest.id
   instance_type               = var.ec2_instance_type
   key_name                    = local.ssh_key_name
-  iam_instance_profile        = data.aws_iam_instance_profile.cwagent_instance_profile.name
-  vpc_security_group_ids      = [data.aws_security_group.ec2_security_group.id]
+  iam_instance_profile        = module.basic_components.instance_profile
+  subnet_id                   = module.basic_components.random_subnet_instance_id
+  vpc_security_group_ids      = [module.basic_components.security_group]
   associate_public_ip_address = true
   get_password_data           = true
+
   metadata_options {
     http_endpoint = "enabled"
     http_tokens   = "required"
   }
-  user_data = <<EOF
-<powershell>
-Write-Output "Install OpenSSH and Firewalls which allows port 22 for connection"
-Add-WindowsCapability -Online -Name OpenSSH.Client~~~~0.0.1.0
-Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
-
-Start-Service sshd
-Set-Service -Name sshd -StartupType 'Automatic'
-
-[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072
-Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
-
-choco install git --confirm
-choco install go --confirm
-msiexec /i https://awscli.amazonaws.com/AWSCLIV2.msi  /norestart /qb-
-
-[Environment]::SetEnvironmentVariable("PATH", "C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files\Amazon\AWSCLIV2\;C:\Program Files\Go\bin;C:\Windows\System32;C:\Windows\System32\WindowsPowerShell\v1.0\", [System.EnvironmentVariableTarget]::Machine)
-</powershell>
-EOF
 
   tags = {
-    Name = "cwagent-integ-test-ec2-windows-${element(split("/", var.test_dir), 3)}-$${module.common.testing_id}"
+    Name = "cwagent-integ-test-ec2-windows-${module.common.testing_id}"
   }
 }
 
 resource "null_resource" "integration_test" {
   depends_on = [aws_instance.cwagent]
+
   # Install software
+  connection {
+    type     = "winrm"
```

Comment:
Instead of using OpenSSH with password which would allow attacker to brute force into it, we would use `winrm` which have [more control in the authentication](https://www.reddit.com/r/sysadmin/comments/nadfbs/comment/gxusz9w/?utm_source=share&utm_medium=web2x&context=3) and be firewall friendly. The WinRM configuration has being built in the EC2 Image pipeline